### DISCUSSION TITLE
Options for Handling Wildcard(*) in Playback Files

### DISCUSSION THREAD

**Discussion by GPayne** [OpenSpace/OpenSpace#2130] [Polls]
[Issue#2121](https://github.com/OpenSpace/OpenSpace/issues/2121) has to do with a session recording containing a script command with a wildcard (*) which can apply to one or multiple scenegraphnodes.

This problem happens because the software scrutinizes all script commands in a recording, and adds new matching initialization commands at the start of the playback file. There are a few reasons why it does this:
1. It's necessary to get correct playback behavior. For example, if a user creates a recording where a planet trail fades-in over time, then playback must start with that trail visibility disabled. If not, the fade would not happen, so a script command to hide that trail visibility gets added to the start of the file. This is also necessary for playback loop mode.
2. Scrutinizing the commands will catch a case where necessary nodes are not currently loaded (e.g. trying to playback a NewHorizons recording when in the default profile).

Since a command with a wildcard is not specific, there are two options that I'm considering for how to handle this. They are described in detail here (select one in the form below):
1. **Ignore Wildcard**: Start the playback even if the command has no matches in the currently-loaded nodes. One reason for doing this is that the user doesn't care if certain nodes are loaded. For example, recording a wildcard  command to disable visibility of all satellites would be insurance against them appearing, whether or not they are currently loaded.
2. **Enforce Wildcard Match**: Refuse to start playback unless there is at least 1 match in the currently-loaded nodes. The assumption here is that if the user recorded this script command, that they must have intended that at least 1 node would be loaded.

**Comment by ylvaselling** [comment]
I vote "ignore with error message"
